WordPress/gutenberg
# [Try] Customizable toolbar contents

## Description

This PR makes the block toolbar contents customizable via a slot and is a first step towards resolving https://github.com/WordPress/gutenberg/issues/23375.

It adds the way to customize toolbar contents, essentially replacing all the default toolbar contents. After implementing the transition between the normal and the customized state, I thought it would be cool if selecting another block would also resize the toolbar smoothly - so now it does that too:

![2020-07-01 15-52-37 2020-07-01 15_53_23](https://user-images.githubusercontent.com/205419/86252070-04a9d300-bbb3-11ea-9f96-d29f04934a20.gif)

## How has this been tested?

This PR only adds the infrastructure. To actually test, you will need to apply both this PR and https://github.com/WordPress/gutenberg/pull/23614. Then:

1. Go to experimental navigation screen
1. Select a link block, switch to another one, select just the text contents, select an entire block.
1. Confirm that the animation worked well.
1. Click on the "Edit Link" button while a link block is selected.
1. Confirm toolbar contents were replaced, the animation was smooth, and the keyboard navigation works.
1. Go off the script, play with it, confirm it works as you would expect it to.

## Screenshots

## Types of changes
New feature (non-breaking change which adds functionality)

## Checklist:
